km/km_main.c

```diff
@@ -636,30 +613,9 @@ int main(int argc, char* argv[])
    int argc_p;      // payload's argc
    char** argv_p;   // payload's argc (*not* in ABI format)
    char* payload_name;
-   char* pl_name;
-
-   km_check_do_shell();
-
-   if (km_do_shell != 0) {
-      /*
-       * Sometimes km is invoked as a payload name that is a symlink to km.
-       * In that case, no km related arguments will be found on the command line.
-       * Detect those cases here and help km_trace_setup() avoid treating payload
-       * arguments as km arguments.
-       * First call works for symlinks, including found in PATH.
-       * But for shebang the first is shebang file itself, hence the second call.
-       */
-      if ((pl_name = km_traverse_payload_symlinks((const char*)getauxval(AT_EXECFN))) == NULL) {
-         pl_name = km_traverse_payload_symlinks((const char*)argv[0]);
-      }
-   } else {
-      pl_name = NULL;
-   }
+   char* pl_name = NULL;
+
    km_trace_setup(argc, argv, pl_name);   // setup trace settings as early as possible
```

Comment:
Since pl_name is always null now, there are probably changes needed in km_trace_setup.
At the least we can get rid of the pl_name variable.

Reply:
Not sure about getting rid of pl_name entirely, looks like we pass it to km_parse_args().

Reply:
Looks like km_parse_args() could use some code removal too if the pl_name argument is always null.
But... I think this change is pretty big and don't want it to get bigger.